internal/command/console.go

```diff
@@ -144,14 +155,16 @@ func (c *ConsoleCommand) Run(args []string) int {
 	// set the ConsoleMode to true so any available console-only functions included.
 	scope.ConsoleMode = true
 
-	if diags.HasErrors() {
-		diags = diags.Append(tfdiags.SimpleWarning("Due to the problems above, some expressions may produce unexpected results."))
-	}
-
 	// Before we become interactive we'll show any diagnostics we encountered
 	// during initialization, and then afterwards the driver will manage any
 	// further diagnostics itself.
-	c.showDiagnostics(diags)
+	if diags.HasErrors() {
```

Comment:
Hmm. :thinking: I think you are right.

What I was trying to fix here is that at some point since we implemented this command we changed `c.showDiagnostics` to first sort the diagnostics into a canonical sort order, and that order always places warnings first, so we were ending rendering "Due to the problems above" in a message that was above all of the problems.

But you're right that the last two lines of this branch should've stayed outside of the condition, since we do always want to print diagnostics at least once. I'll push a fix shortly!